### PR TITLE
make function more readable

### LINKED ISSUE #93744: scheduler PreFilter NodePorts plugins  clean up getContainerPorts Loop  naming
When I learned more about the entire kubernetes scheduler frame from the source code.

I noticed that this is the getContainerPort function of nodeport plugins in the PreFilter stage.

This is the original code as following as bellowing.It is seems to me  that the naming and use of for loops is not very good for reading 
```go
func getContainerPorts(pods ...*v1.Pod) []*v1.ContainerPort {
	ports := []*v1.ContainerPort{}
	for _, pod := range pods {
		for j := range pod.Spec.Containers {
			container := &pod.Spec.Containers[j]
			for k := range container.Ports {
				ports = append(ports, &container.Ports[k])
			}
		}
	}
	return ports
}
```
how about let's revise the naming as bellowing.
```go
func getContainerPorts(pods ...*v1.Pod) []*v1.ContainerPort {
	ports := []*v1.ContainerPort{}
	for _, pod := range pods {
		for _, container := range pod.Spec.Containers {
			for k := range container.Ports{
				ports = append(ports, &container.Ports[k])
			}
		}
	}
	return ports
}
```

### PR DESCRIPTION
**Pull request by alam0rt**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

This PR makes the  _getContainerPorts()_ function more readable

**Which issue(s) this PR fixes**:

Fixes #93744

**Special notes for your reviewer**:

/cc @jjmengze

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/nodeports/node_ports.go [hunk 1]
```diff
@@ -61,10 +61,10 @@ func (pl *NodePorts) Name() string {
 func getContainerPorts(pods ...*v1.Pod) []*v1.ContainerPort {
 	ports := []*v1.ContainerPort{}
 	for _, pod := range pods {
-		for j := range pod.Spec.Containers {
-			container := &pod.Spec.Containers[j]
-			for k := range container.Ports {
-				ports = append(ports, &container.Ports[k])
+		for _, container := range pod.Spec.Containers {
```
Comment (habibrosyad) [inline pkg/scheduler/framework/plugins/nodeports/node_ports.go:64]: I think this is a duplicate of https://github.com/kubernetes/kubernetes/pull/93745 which has already received several feedbacks.

Comment (alam0rt) [reply pkg/scheduler/framework/plugins/nodeports/node_ports.go]: Thanks for your feedback! I have tried to take the point about not wanting to create _new_ references into consideration. Not sure if this is any more readable.

File: pkg/scheduler/framework/plugins/nodeports/node_ports.go [hunk 2]
```diff
@@ -64,7 +64,8 @@ func getContainerPorts(pods ...*v1.Pod) []*v1.ContainerPort {
 		for j := range pod.Spec.Containers {
 			container := &pod.Spec.Containers[j]
 			for k := range container.Ports {
-				ports = append(ports, &container.Ports[k])
+				port := &container.Ports[k]
+				ports = append(ports, port)
```
Comment (Huang-Wei) [inline pkg/scheduler/framework/plugins/nodeports/node_ports.go:68]: I don't quite think it's more readable - `&container.Ports[k]` is already simple enough.

File: pkg/scheduler/framework/plugins/nodeports/node_ports.go (hunk 1 above)
Comment (Huang-Wei) [reply pkg/scheduler/framework/plugins/nodeports/node_ports.go]: Not worth introducing this change IMO: https://github.com/kubernetes/kubernetes/pull/93823#discussion_r468085964

File: pkg/scheduler/framework/plugins/nodeports/node_ports.go (hunk 2 above)
Comment (alam0rt) [reply pkg/scheduler/framework/plugins/nodeports/node_ports.go:68]: Yeah I feel you're right. Closing

**Pull request closed without merging**